### DISCUSSION TITLE
How does the D2 docs site show D2 code blocks with syntax highlights?

### DISCUSSION THREAD

**Discussion by Airkro** [terrastruct/d2#1488] [Q&A]
Apparently, it's not implemented using Prismjs that comes with Docusaurus.

**Comment by alixander** [comment]
@Airkro we implement our own component that highlights with vscode syntax highlighters: https://github.com/terrastruct/d2-docs/blob/master/src/theme/CodeBlock/index.js

docusaurus picks up this special component due to their "swizzling" mechanism: https://docusaurus.io/docs/swizzling#which-component-should-i-swizzle

--------


it'd be nice to get something in prismjs, but i just looked and they're not accepting PRs until later: https://github.com/PrismJS/prism#important-notice
